api/v1alpha3/awscluster_types.go

```diff
@@ -148,6 +152,23 @@ type AWSClusterStatus struct {
 	Conditions     clusterv1.Conditions     `json:"conditions,omitempty"`
 }
 
+type S3Bucket struct {
+	// Enabled controls if S3 Bucket should be created as part of cluster provisioning.
+	Enabled bool `json:"enabled,omitempty"`
+
+	// ControlPlaneIAMInstanceProfile is a name of the IAMInstanceProfile, which will be allowed
+	// to read control-plane node bootstrap data from S3 Bucket.
+	ControlPlaneIAMInstanceProfile string `json:"controlPlaneIAMInstanceProfile,omitempty"`
+
+	// NodesIAMInstanceProfiles is a list of IAM instance profiles, which will be allowed to read
+	// worker nodes bootstrap data from S3 Bucket.
+	NodesIAMInstanceProfiles []string `json:"nodesIAMInstanceProfiles,omitempty"`
+
+	// Name defines name of S3 Bucket to be created. By default Cluster name will be used.
```

Comment:
I think something more should be used for the naming. Some naming scheme which includes `cluster-api` with the name. We do something similar in our current implementation and apply the name to a tag as well (to easily filter by cluster).

Reply:
This can be changed in S3 service implementation. I'm open for suggestions, as I'm not really familiar with naming convention in this project.

Reply:
Changed to `[namespace]_[clustername]` as a default name then.

Reply:
Marking as resolved. Bucket name is a now a required parameter.